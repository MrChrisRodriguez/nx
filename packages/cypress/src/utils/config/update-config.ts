import {
  createPrinter,
  createSourceFile,
  ScriptTarget,
  transform,
} from 'typescript';
import {
  CypressConfigPt2,
  mergeCypressConfigs,
} from './add-update-property-transformer';

// TODO(caleb): handle devserver in config
export function addOrUpdateConfigProperties(
  configContent: string,
  config: CypressConfigPt2,
  overwrite: boolean = false
) {
  const sourceFile = createSourceFile(
    'cypress.config.ts',
    configContent,
    ScriptTarget.Latest,
    true
  );

  const transformedResult = transform(sourceFile, [
    mergeCypressConfigs(config, overwrite),
  ]);

  return createPrinter().printFile(transformedResult.transformed[0]);
}

export function removeConfigProperties() {}

// export function addConfigImport() {}
//
// export function removeConfigImport() {}
