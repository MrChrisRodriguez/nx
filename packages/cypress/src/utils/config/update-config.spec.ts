import { addOrUpdateConfigProperties } from './update-config';
import { componentDevServer } from '@nrwl/cypress/plugins/react';

describe('Update Cypress Config', () => {
  let configContent = `
import { defineConfig } from 'cypress';
import { componentDevServer } from '@nrwl/cypress/plugins/next';


export default defineConfig({
  baseUrl: 'blah its me',
  component: {
    devServer: componentDevServer('tsconfig.cy.json', 'babel'),
    pluginsFile: false,
    video: true,
    chromeWebSecurity: false,
    fixturesFolder: 'cypress/fixtures',
    specPattern: '**/*.cy.{js,jsx,ts,tsx}',
    supportFile: 'cypress/support/component.ts',
    videosFolder: '../../dist/cypress/apps/n/videos',
    screenshotsFolder: '../../dist/cypress/apps/n/screenshots',
  },
  e2e: {
    fileServerFolder: '.',
    fixturesFolder: './src/fixtures',
    integrationFolder: './src/e2e',
    supportFile: './src/support/e2e.ts',
    specPattern: '**/*.cy.{js,ts}',
    pluginsFile: false,
    video: true,
    videosFolder: '../../dist/cypress/apps/myapp4299814-e2e/videos',
    screenshotsFolder: '../../dist/cypress/apps/myapp4299814-e2e/screenshots',
    chromeWebSecurity: false,
  }
});
  `;

  describe('Properties', () => {
    it('should add a property', () => {
      const actual = addOrUpdateConfigProperties(configContent, {
        blah: 'i am a top level property',
        baseUrl: 'http://localhost:1234',
        component: {
          fixturesFolder: 'cypress/fixtures/cool',
          devServer: componentDevServer('tsconfig.cy.json', 'swc'),
          // @ts-ignore
          blah: 'i am a random property',
        },
        e2e: {
          video: false,
        },
      });

      expect(actual).toMatchSnapshot();
    });

    it('should update a property', () => {});

    it('should update a nested property', () => {});

    it('should remove a property', () => {});

    it('should remove a nested property', () => {});
  });
});
