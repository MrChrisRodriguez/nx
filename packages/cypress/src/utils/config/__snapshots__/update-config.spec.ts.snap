// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Cypress Config Properties should add a property 1`] = `
"import { defineConfig } from 'cypress';
import { componentDevServer } from '@nrwl/cypress/plugins/next';
export default defineConfig({
    blah: 'i am a top level property',
    baseUrl: 'http://localhost:1234',
    component: {
        devServer: componentDevServer('tsconfig.cy.json', 'babel'),
        pluginsFile: false,
        video: true,
        chromeWebSecurity: false,
        fixturesFolder: 'cypress/fixtures/cool',
        specPattern: '**/*.cy.{js,jsx,ts,tsx}',
        supportFile: 'cypress/support/component.ts',
        videosFolder: '../../dist/cypress/apps/n/videos',
        screenshotsFolder: '../../dist/cypress/apps/n/screenshots',
        blah: 'i am a random property'
    },
    e2e: {
        fileServerFolder: '.',
        fixturesFolder: './src/fixtures',
        integrationFolder: './src/e2e',
        supportFile: './src/support/e2e.ts',
        specPattern: '**/*.cy.{js,ts}',
        pluginsFile: false,
        video: false,
        videosFolder: '../../dist/cypress/apps/myapp4299814-e2e/videos',
        screenshotsFolder: '../../dist/cypress/apps/myapp4299814-e2e/screenshots',
        chromeWebSecurity: false,
    }
});
"
`;
